tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -486,6 +489,131 @@ def validate_executor_placements(cls, executor_placements):
                   'Unsupported cardinality for placement "{}": {}.'.format(
                       pl, pl_cardinality))
 
+  async def _trusted_aggregator_generate_keys(self):
+    
+    @computations.tf_computation()
+    def generate_keys():
+      pk, sk = easy_box.gen_keypair()
+      return pk.raw, sk.raw
+
+    fn_type = generate_keys.type_signature
+    fn = generate_keys._computation_proto
+
+    aggregator_ex = self._get_child_executors(placement_literals.AGGREGATOR,
+                                              index=0)
+
+    result = await aggregator_ex.create_call(
+        await aggregator_ex.create_value(fn, fn_type)
+    )
+
+    pk = await aggregator_ex.create_selection(result, 0)
+    sk = await aggregator_ex.create_selection(result, 1)
+
+    pk_fed_val = await self._place(pk, placement_literals.AGGREGATOR)
```

Comment:
why do you need the two `_place` calls? isn't it enough that you used `aggregator_ex` above?

Reply:
ah good question. I have done this because `result` (calls output) is an `EagerValue`. So no formal placement: `result.type_signature == NamedTupleType([TensorType(tf.uint8, [32]), TensorType(tf.uint8, [32])])`

I did `_place` on aggregator to create a `FederatedValue` with formal placement: `FederatedType(TensorType(tf.uint8, [32]), PlacementLiteral('aggregator'), True)`

Reply:
I see. There's some redundancy here since the values already live in the aggregator, and calling `_place` essentially copies them from the driver back into it. A better solution would be to either add a check to `_place` that skips copying if already on the desired placement, or better, avoid `_place` in your code and simply add the missing placement manually.

We can leave this as future optimization.